src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -25,22 +27,75 @@ public class DeleteCommand extends Command {
 
     public static final String MESSAGE_DELETE_CONTACT_SUCCESS = "Deleted Contact: %1$s";
 
-    private final Index targetIndex;
+    private final Object selector; // TODO: This is very sus but this will only be used for equals comparison
```

Comment:
As a temporary solution that comparatively isn't as questionable, we can use two separate variables for `Index` and `Id` with `null` values if they aren't set.

If this is for the purposes of extensibility, given that it's `private`, it means you'd need to modify the code here anyways, of which it doesn't really matter whether you store it in the same variable or different ones. Hence, I think it's ok regardless if this is a single `Object` or two separate variables.